app/controllers/api/users.php

```diff
@@ -1420,6 +1422,235 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
+    ->label('sdk.description', '/docs/references/users/create-session.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_SESSION)
+    ->param('userId', '', new CustomId(), 'User ID. Choose a custom ID or generate a random ID with `ID.unique()`. Valid chars are a-z, A-Z, 0-9, period, hyphen, and underscore. Can\'t start with a special char. Max length is 36 chars.')
+    ->inject('request')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('locale')
+    ->inject('geodb')
+    ->inject('queueForEvents')
+    ->action(function (string $userId, Request $request, Response $response, Database $dbForProject, Document $project, Locale $locale, Reader $geodb, Event $queueForEvents) {
+        $user = $dbForProject->getDocument('users', $userId);
+
+        if ($user !== false && !$user->isEmpty()) {
+            $user->setAttributes($user->getArrayCopy());
+        } else {
+            $limit = $project->getAttribute('auths', [])['limit'] ?? 0;
+
+            if ($limit !== 0) {
+                $total = $dbForProject->count('users', max: APP_LIMIT_USERS);
+
+                if ($total >= $limit) {
+                    throw new Exception(Exception::USER_COUNT_EXCEEDED);
+                }
+            }
+
+            if ($userId !== 'unique()') {
+                $existingUser = $dbForProject->findOne('users', [
+                    Query::equal('$id', [$userId]),
+                ]);
+
+                if ($existingUser !== false && !$existingUser->isEmpty()) {
+                    throw new Exception(Exception::USER_ALREADY_EXISTS);
+                }
+            }
+
+            $userId = $userId === 'unique()' ? ID::unique() : $userId;
+
+            $user->setAttributes([
+                '$id' => $userId,
+                '$permissions' => [
+                    Permission::read(Role::any()),
+                    Permission::update(Role::user($userId)),
+                    Permission::delete(Role::user($userId)),
+                ],
+                'email' => null,
+                'emailVerification' => false,
+                'status' => true,
+                'password' => null,
+                'hash' => Auth::DEFAULT_ALGO,
+                'hashOptions' => Auth::DEFAULT_ALGO_OPTIONS,
+                'passwordUpdate' => null,
+                'registration' => DateTime::now(),
+                'reset' => false,
+                'prefs' => new \stdClass(),
+                'sessions' => null,
+                'tokens' => null,
+                'memberships' => null,
+                'search' => implode(' ', [$userId]),
+                'accessedAt' => DateTime::now(),
+            ]);
+
+            $user->removeAttribute('$internalId');
+            Authorization::skip(fn () => $dbForProject->createDocument('users', $user));
+        }
```

Comment:
This is a `createSession` endpoint right? I see this is performing 2 functions
1. Creating a user 
2. Creating a session for the user

this goes against REST principles where each endpoint should handle only one responsibility unless absolutely necessary. 

What is the need to create a user here when we already have the `createUser` endpoint?

This `createSession` endpoint should return 404 if the user does not exist

Reply:
I agree it seems to violate REST principle, but it offers pontentially better DX, and Matej proposed this change justifying it as the existing behaviour for endpoints in account service.

Should we:
1. Auto-create users for endpoints in account service and NOT users service?
2. Auto-create users for endpoints in both account and users services?
3. Change accounts service so we never auto-create users?